### ISSUE TITLE
Blog: image file gets deleted

### ISSUE THREAD

**Issue by megasnort** [forkcms/forkcms#1114]
Funny things happen when images are added to different versions of blogposts. I already have a solution (already implemented in a project), but want to check if you agree:

**case 1**
Problem: When upload a new picture, the previous one is deleted. When you reactivate a previous version of the blogpost, the image is not present anymore (but is in the database)
Solution: just don't delete the previous image

**case 2**
Problem: When you don't upload a new picture, the existing picture gets renamed. When you then reactivate a previous post, the image is not present (because is was renamed)
Solution 1: Don't rename the picture. The small price to pay is that the SEO (filename) of this image might be a little bit less relevant.
Solution 2: The other option is the copy to file every time the newly generated filename is different

**case 3**
Problem: When you check 'delete image', the image is deleted. If the image was used in previous versions, it isn't available there anymore.
Solution: just don't delete the previous image

Bottom line: The image should only be deleted when the post itself gets deleted.

**Comment by WouterSioen** [comment]
I fully agree with your concerns and your idea's.

I think it could be easier though, to put each image in a folder with as id the revision_id of the post. This way, you'll be able to rename the file for SEO, and it will be really easy to remove the image if the revision is deleted (just delete the folder).

**Comment by megasnort** [comment]
Separate revisions cannot be deleted, so that might not be necessary.

**Comment by WouterSioen** [comment]
Don't they get cleaned up after a while? I thought only a limited number of versions was kept.

**Comment by megasnort** [comment]
I now do have the ultimate solution: With the Symfony\Component\Filesystem\Filesystem you can create a symlink. Ideal for saving diskspace and SEO friendly.

Also came across another problem, when adding a new file without changing the title of the blogpost, the previous image gets overwritten.

So I'll have to add something extra there.

**Comment by WouterSioen** [comment]
I don't think using symlinks is the ultimate solution, since not all types of hosting will allow PHP to create symlinks.

**Comment by megasnort** [comment]
the symlink function has a parameter for that:

  public function symlink($originDir, $targetDir, $copyOnWindows = false)

Or are symlinks not supported on other Fork-platforms as well? I actually don't know.

**Comment by WouterSioen** [comment]
If you use cheap hosting, you will never be able to create symlinks, even on linux based hosting. We don't use symlinks in Fork for now for that reason.

**Comment by megasnort** [comment]
Ok, I'll use the copy-function. There are not so many cases where a file has to be copied, only when the meta-url changes.

**Comment by megasnort** [comment]
And I don't see any code that deletes versions.

**Comment by WouterSioen** [comment]
https://github.com/forkcms/forkcms/blob/master/src/Backend/Modules/Blog/Engine/Model.php#L1028-L1051

**Comment by megasnort** [comment]
My bad, i was searching for 'versions' :)

Anyway, images do not seem to get deleted there too, so another thing to fix.
